numba/np/arrayobj.py

```diff
@@ -1303,6 +1303,76 @@ def _broadcast_to_shape(context, builder, arrtype, arr, target_shape):
     return new_arrtype, new_arr
 
 
+@intrinsic
+def _numpy_broadcast_to(typingctx, array, shape):
+    ret = types.Array(array.dtype, shape.count, array.layout, readonly=True)
```

Comment:
From the debugging call:

```suggestion
    ret = types.Array(array.dtype, shape.count, 'A', readonly=True)
```

This will prevent strides being ignored when indexing into the broadcasted array (which happens when the layout is `'C'`)

Reply:
Done! I've put both you and Brandon as co-authors in the commit. Hope that's ok